conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
```

Comment:
Yeah, `bytes` is defined on both and on Python 2 `bytes == str`, but not on Python 3. I could be wrong about what you are trying to accomplish here, but it seems the patch will have to change anyways as `sys.getfilesystemencoding()` doesn't seem to work for Windows.